src/Hl7.Fhir.Core.Tests/Rest/SearchAsyncTests.cs

```diff
@@ -66,20 +67,37 @@ public async Task SearchUsingPost_UsingSearchParams_SearchReturned()
             var client = new LegacyFhirClient(_endpointSupportingSearchUsingPost);
             client.Settings.PreferredFormat = ResourceFormat.Json;
             client.Settings.PreferredReturn = Prefer.ReturnRepresentation;
-            
+
             await searchUsingPost(client);
         }
 
         [TestMethod]
         [TestCategory("IntegrationTest")]
         public async Task SearchUsingPost_UsingSearchParams_SearchReturnedHttpClient()
         {
-            using (var client = new FhirClient(_endpoint)) 
+            using (var client = new FhirClient(_endpoint))
             {
                 client.Settings.PreferredFormat = ResourceFormat.Json;
                 client.Settings.PreferredReturn = Prefer.ReturnRepresentation;
                 await searchUsingPost(client);
-            }           
+            }
+        }
+
+        [TestMethod]
+        public async Task SearchUsingPost_Issue1591()
```

Comment:
This file probably should not be part of this PR?

Reply:
Oops, you are right. I will fix it right away. Good catch!